wrangler-service/src/main/java/io/cdap/wrangler/service/directive/WranglerDisplaySerializer.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.wrangler.service.directive;
+
+import com.google.gson.Gson;
+import com.google.gson.TypeAdapter;
+import com.google.gson.TypeAdapterFactory;
+import com.google.gson.reflect.TypeToken;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonWriter;
+import io.cdap.wrangler.api.Pair;
+import io.cdap.wrangler.api.Row;
+import org.apache.commons.lang3.ClassUtils;
+
+import java.io.IOException;
+import java.nio.Buffer;
+
+/**
+ * Formats Structure and Array records in a "pseudojson" in which it collapses
+ * complex types by callint toString().  This is done so the complex types are shown
+ * in a similar way inside and outside the structure records.
+ */
+public class WranglerDisplaySerializer implements TypeAdapterFactory {
+
+  public static final String NONDISPLAYABLE_STRING = "Non-displayable object";
+
+  @Override
+  public <T> TypeAdapter<T> create(Gson gson, TypeToken<T> type) {
+    // Let GSON Handle primitives and collections
+    if (ClassUtils.isPrimitiveOrWrapper(type.getRawType())
+      || (Iterable.class.isAssignableFrom(type.getRawType()))) {
+      return null;
+    }
+    if (Row.class.isAssignableFrom(type.getRawType())) {
+      return (TypeAdapter<T>) rowClassAdapter(gson);
+    }
+
+    if (Buffer.class.isAssignableFrom(type.getRawType())) {
```

Comment:
What is the current behavior of dealing Buffer, are we showing the actual bytes value? This change will be backward incompatible.

Reply:
As mentioned offline. This mirrors current behavior.

Reply:
The original code does not have any special handling of the Bytes type and still show it non-displayable, I don't think we need this check for special handling of Buffer, since all Buffer will be converted to byte[] in https://github.com/data-integrations/wrangler/blob/develop/wrangler-core/src/main/java/io/cdap/wrangler/utils/StructuredToRowTransformer.java#L81-L83. Let's remove this check and add it back when we support other Buffer type in the future.